docs/reference/setup/advanced-configuration.asciidoc

```diff
@@ -149,3 +149,25 @@ options. We do not recommend using `ES_JAVA_OPTS` in production.
 
 NOTE: If you are running {es} as a Windows service, you can change the heap size
 using the service manager. See <<windows-service>>.
+
+[[readiness-tcp-port]]
+===== Enable the Elasticsearch TCP readiness port
+
+NOTE: This feature is currently experimental for internal use by Elastic software only.
+
+If configured, a node can open a TCP port when the node is in a ready state. A node is deemed
+ready when it has successfully joined a cluster. In a single node configuration, the node is
+said to be ready, when it's able to accept requests.
```

Comment:
```suggestion
When the readiness TCP port is enabled, a client can connect to the port to determine whether or not the {es} node is in a ready state. 
A node is in a ready state once it has successfully joined a cluster. 
In a single node cluster, the node becomes ready once it can  accept requests.
```